File: src/Programmers/P1/P1_12947.java
package Programmers.P1;


public class P1_12947 {
    public static void main(String[] args) {
        System.out.println(solution(18));
    }

    public static boolean solution(int x) {
        int sum = 0;
        int temp = x;
        while (temp != 0) {
            sum += temp % 10;
            temp = temp / 10;
        }
        return x % sum == 0;
    }
}

File: src/programmers/p1/P1_12947.kt
package programmers

fun main() {
    println(solution(10))
    println(solution2(12))
    println(solution(11))
    println(solution2(13))
}

private fun solution(x: Int): Boolean = x % x.toString().map { v -> Character.getNumericValue(v) }.reduce { a, b -> a + b } == 0
private fun solution2(x: Int): Boolean = x % x.toString().map { v -> Character.getNumericValue(v) }.sum() == 0
